controllers/crd_controller.go

```diff
@@ -0,0 +1,92 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"github.com/flanksource/kommons"
+	"github.com/flanksource/template-operator/k8s"
+	"github.com/go-logr/logr"
+	apiv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// CRDReconciler reconciles changes to CRD objects
+type CRDReconciler struct {
+	ControllerClient client.Client
+	Client           *kommons.Client
+	Events           record.EventRecorder
+	Log              logr.Logger
+	Scheme           *runtime.Scheme
+	Cache            *k8s.SchemaCache
+	CRDcache		 map[string]string
+
+}
+
+// +kubebuilder:rbac:groups="*",resources="*",verbs="*"
+
+func (r *CRDReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("crd", req.NamespacedName)
+	log.V(2).Info("crd update detected, checking cache state")
+	crd := &apiv1.CustomResourceDefinition{}
+
+	if err := r.ControllerClient.Get(context.Background(), req.NamespacedName, crd); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	resourceVersion := crd.ResourceVersion
+
+	if version, ok := r.CRDcache[req.NamespacedName.String()]; ok {
+		if resourceVersion != version {
```

Comment:
I think that `resourceVersion` is global and it's incremented every time an object is created updated. You will need to check if `resourceVersion > version`. Also, I think you can use generation if you are going to check per crd or if you are going to use resourceVersion then you can have a global version field, not for each crd.